### PR TITLE
Re-enable app-build-suite

### PR DESCRIPTION
**Pull request by mcbenjemaa**
This PR:

- adds/changes/removes etc

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Test on Workload cluster.

CC @uvegla

### CODE REVIEW

**Comment by QuentinBisson** [comment]
It's failing validation/ What's the point of having an empty helper? 

```
2022-11-09 09:04:24,368 step_exec_lib.steps INFO: Running pre-run step for HelmBuilderValidator
2022-11-09 09:04:24,368 step_exec_lib.steps INFO: Running pre-run step for GiantSwarmHelmValidator
2022-11-09 09:04:24,370 app_build_suite.build_steps.helm INFO: Running Giant Swarm validator 'C0001: HasTeamLabel'.
2022-11-09 09:04:24,372 helm INFO: The expected team label not found in './helm/prometheus-agent/templates/_helpers.tpl'.
2022-11-09 09:04:24,372 helm INFO: './helm/prometheus-agent/templates/_helpers.tpl' must contain a line that matches the regexp 're.compile('[ \\t]*application\\.giantswarm\\.io/team:[ \\t]+{{[ \\t]*index[ \\t]+\\.Chart\\.Annotations[ \\t]+"application\\.giantswarm\\.io/team"[ \\t]*\\|[ \\t]*quote[ \\t]*}}[ \\t]*')'.
2022-11-09 09:04:24,372 step_exec_lib.steps ERROR: Error when running pre-run step for GiantSwarmHelmValidator: Giant Swarm validator 'C0001: HasTeamLabel' failed its checks.
2022-11-09 09:04:24,372 step_exec_lib.steps ERROR: Error when running pre-steps: Source: 'GiantSwarmHelmValidator', message: Giant Swarm validator 'C0001: HasTeamLabel' failed its checks... Exiting.

Exited with code exit status 1
```

**Comment by uvegla** [comment]
> It's failing validation/ What's the point of having an empty helper?

It should not be empty, cos I think what is in the sub-chart should simply be the actual chart.

File: helm/prometheus-agent/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{{/*
+Common labels
+*/}}
+{{- define "labels.common" -}}
+{{ include "labels.selector" . }}
+app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+app.kubernetes.io/version: {{ .Values.image.tag | default .Chart.AppVersion | quote }}
+application.giantswarm.io/team: {{ index .Chart.Annotations "application.giantswarm.io/team" | quote }}
+helm.sh/chart: {{ include "chart" . | quote }}
+{{- end -}}
```
Comment (QuentinBisson) [inline helm/prometheus-agent/templates/_helpers.tpl:32]: Can you remove thé eol to make Théo happy?

**Comment by QuentinBisson** [comment]
@mcbenjemaa you can use this as an example to fix your issue:
https://github.com/giantswarm/prometheus-operator-app/blob/daad39924e2623cd9f8d225deb2328f0185f75a1/.kube-linter.yaml

You only need dangling-service I think

**Comment by QuentinBisson** [comment]
@uvegla @mcbenjemaa can you check this?

**Comment by hervenicol** [comment]
Some (non-blocking) concerns about this chart:
* we are using a chart in a chart, so that we could someday release a generic prometheus-agent chart as OSS.
* meaning any giantswarm-specific stuff should be in the top-level chart, and the sub-chart should be as generic as possible.
* WRT this, `giantswarm.io/team` label should be defined in the top-level chart's helpers.
* On top of that, defining it in the top-level chart would ensure it gets used by top-level additional templates as well.

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**